Dockerfile

```diff
@@ -1,256 +1,84 @@
 FROM ubuntu
 
-# Install rust
+# Set up essential build tools and libraries
+RUN apt-get update && apt-get install -y -qq --no-install-recommends \
```

Comment:
This makes it hard to keep track of which deps are needed by which lib, I'd prefer to keep them separate, or keep the single command but indicate with comments where each dep "comes from".